core/domain/exp_services.py

```diff
@@ -359,6 +359,240 @@ def export_states_to_yaml(exploration_id, version=None, width=80):
     return exploration_dict
 
 
+def apply_change_list_to_exploration(
+    old_exploration, current_version, change_list
+):
+    """Applies a changelist to a pristine exploration and returns the result.
```

Comment:
I think its because the `old_exploration` should be in the original condition and unchanged before passing it to this function. Also, this part was earlier inside `apply_change_list` of `exp_services`. I have just refactored it so that this part of the code can also be used by the beam job.

Reply:
"pristine" is a bit hard to understand, it is not common word, can we use something more understandable?